pkg/operator/resources/batchapi/k8s_specs.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2020 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package batchapi
+
+import (
+	"path"
+
+	"github.com/cortexlabs/cortex/pkg/lib/hash"
+	"github.com/cortexlabs/cortex/pkg/lib/k8s"
+	"github.com/cortexlabs/cortex/pkg/lib/pointer"
+	"github.com/cortexlabs/cortex/pkg/operator/config"
+	"github.com/cortexlabs/cortex/pkg/operator/operator"
+	"github.com/cortexlabs/cortex/pkg/types/spec"
+	"github.com/cortexlabs/cortex/pkg/types/userconfig"
+	istioclientnetworking "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	kbatch "k8s.io/api/batch/v1"
+	kcore "k8s.io/api/core/v1"
+)
+
+const _operatorService = "operator"
+
+func k8sJobSpec(api *spec.API, job *spec.Job) (*kbatch.Job, error) {
+	switch api.Predictor.Type {
+	case userconfig.TensorFlowPredictorType:
+		return tensorFlowPredictorJobSpec(api, job)
+	case userconfig.ONNXPredictorType:
+		return onnxPredictorJobSpec(api, job)
+	case userconfig.PythonPredictorType:
+		return pythonPredictorJobSpec(api, job)
+	default:
+		return nil, nil // unexpected
+	}
+}
+
+func pythonPredictorJobSpec(api *spec.API, job *spec.Job) (*kbatch.Job, error) {
+	containers, volumes := operator.PythonPredictorContainers(api)
+	for i, container := range containers {
+		if container.Name == operator.APIContainerName {
+			containers[i].Env = append(container.Env, kcore.EnvVar{
+				Name:  "CORTEX_JOB_SPEC",
+				Value: "s3://" + config.Cluster.Bucket + "/" + job.FileSpecKey(),
+			})
+		}
+	}
+
+	return k8s.Job(&k8s.JobSpec{
+		Name:        job.JobKey.K8sName(),
```

Comment:
The name `JobKey` could be misleading (due to S3 Keys), but I can't think of a better proposal right now. We can discuss